### PR TITLE
fix: allow serverless plugin empty user-agent

### LINKED ISSUE #9351: bug: proxy-rewrite does not work with aws-lambda plugin
### Current Behavior

When configuring an ApisixRoute endpoint, the `proxy-rewrite` allows me to for example remove specific request headers.

This works fine, unless the request is routed to AWS Lambda function through `aws-lambda` plugin.

The destination function get triggered successfully, yet the headers are not removed.

### Expected Behavior

`proxy-rewrite` should work the same regardless of the final endpoint

### Error Logs

Some header manipulation code IS running, as can be seen from apisix logs printed while processing request:
```
apisix-5b49f6b74-wvl4r apisix 2023/04/21 09:08:49 [warn] 48#48: *9023948 [lua] request.lua:120: set_header(): DEPRECATED: use set_header(ctx, header_name, header_value) instead, client: 10.200.160.150, server: _, request: "POST /lbiegaj/test HTTP/1.1", host: "api.example.com"
apisix-5b49f6b74-wvl4r apisix 2023/04/21 09:08:49 [warn] 48#48: *9023948 [lua] request.lua:120: set_header(): DEPRECATED: use set_header(ctx, header_name, header_value) instead, client: 10.200.160.150, server: _, request: "POST /lbiegaj/test HTTP/1.1", host: "api.example.com"
apisix-5b49f6b74-wvl4r apisix 2023/04/21 09:08:49 [warn] 48#48: *9023948 [lua] request.lua:120: set_header(): DEPRECATED: use set_header(ctx, header_name, header_value) instead, client: 10.200.160.150, server: _, request: "POST /lbiegaj/test HTTP/1.1", host: "api.example.com"
apisix-5b49f6b74-wvl4r apisix 2023/04/21 09:08:49 [warn] 48#48: *9023948 [lua] request.lua:120: set_header(): DEPRECATED: use set_header(ctx, header_name, header_value) instead, client: 10.200.160.150, server: _, request: "POST /lbiegaj/test HTTP/1.1", host: "api.example.com"
```

### Steps to Reproduce

1. Create an AWS Lambda with function url. Sample code that reflects the received event:
```export const handler = async(event) => {
    const response = {
        statusCode: 200,
        body: JSON.stringify(event),
    };
    return response;
};
```
2. Configure an ApisixRoute that uses the `proxy-rewrite` and `aws-lambda` plugins.
```
spec:
  http:
    backends:
    - serviceName: lambda
      servicePort: 80
    match:
      methods:
      - POST
      paths:
      - /test/*
    name: test
    plugins:
    - config:
        function_uri: https://xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx.lambda-url.eu-west-3.on.aws/
        ssl_verify: false
        timeout: 10000
      enable: true
      name: aws-lambda
    - config:
        headers:
          remove:
          - authorization
          - User-Agent
      enable: true
      name: proxy-rewrite
``` 
4. Send requests and observe that Authorization header is not removed.

### Environment

- APISIX version (run `apisix version`): 3.2.0
- Operating system (run `uname -a`): from official docker image (Linux apisix-5b49f6b74-jkxft 5.10.173-154.642.amzn2.x86_64 #1 SMP Wed Mar 15 00:26:42 UTC 2023 x86_64 GNU/Linux)

Helm charts versions:
- apisix 1.2.0
- apisix-ingress-controller 0.11.3
- etcd 8.7.6

### PR DESCRIPTION
**Pull request by Sn0rt**
### Description

Fixes #9351

### Checklist

- [ ] I have explained the need for this PR and the problem it solves
- [ ] I have explained the changes or the new features added to this PR
- [ ] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [ ] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

**Comment by Sn0rt** [comment]
@soulbird pls help to review the design

File: apisix/plugins/serverless/generic-upstream.lua [hunk 1]
```diff
@@ -35,6 +35,7 @@ return function(plugin_name, version, priority, request_processor, authz_schema,
             function_uri = {type = "string"},
             authorization = authz_schema,
             timeout = {type = "integer", minimum = 100, default = 3000},
+            use_default_user_agent = {type = "boolean", default = true},
```
Comment (soulbird) [inline apisix/plugins/serverless/generic-upstream.lua:38]: I think we can set the default value to false, which can reduce the cost of users.

[the remark this answers is not in the document]
Comment (Sn0rt) [reply apisix/plugins/serverless/generic-upstream.lua:38]: sounds reasonable. the reason to set the default value as true, which is the original logic of the program.

[the remark this answers is not in the document]
Comment (Sn0rt) [reply apisix/plugins/serverless/generic-upstream.lua]: fixed

**Comment by monkeyDluffy6017** [comment]
is this pr still needed?

**Comment by monkeyDluffy6017** [comment]
I will close this pr, feel free to reopen it

**Pull request closed without merging**